tools/doxygen.bzl

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2022 Swift Navigation Inc.
+# Contact: Swift Navigation <dev@swift-nav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+load("//tools:configure_file.bzl", "configure_file_impl")
+
+def _list_to_string(in_list):
+    string = ""
+    first_dir = True
+
+    for src_dir in in_list:
+        if first_dir:
+            first_dir = False
+        else:
+            string += '" "'
+        string += src_dir
+    return string
+
+def _swift_doxygen_impl(ctx):
+    vars = ctx.attr.vars | {}
```

Comment:
This is needed in order to copy dict instead of referencing it. Without this, I received:
`Error: trying to mutate a frozen dict value`

Reply:
I'd add this as a comment in the code